### PR TITLE
MSVC: Fix error in debug mode (Fix #16245)

### LINKED ISSUE #16245: [msvc] number of sections exceeded object file format limit: compile with /bigobj
Only happening in Debug mode

```

"C:\Sources\bitcoin\build_msvc\bitcoin.sln" (default target) (1) ->
"C:\Sources\bitcoin\build_msvc\libbitcoinconsensus\libbitcoinconsensus.vcxproj" (default target) (2) ->
(Lib target) ->
  sha256_sse4.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation
 that consumes this library [C:\Sources\bitcoin\build_msvc\libbitcoinconsensus\libbitcoinconsensus.vcxproj]


"C:\Sources\bitcoin\build_msvc\bitcoin.sln" (default target) (1) ->
"C:\Sources\bitcoin\build_msvc\testconsensus\testconsensus.vcxproj" (default target) (3) ->
"C:\Sources\bitcoin\build_msvc\libbitcoin_util\libbitcoin_util.vcxproj" (default target) (4) ->
  sync.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operation that c
onsumes this library [C:\Sources\bitcoin\build_msvc\libbitcoin_util\libbitcoin_util.vcxproj]
  compat_strnlen.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used by any link operat
ion that consumes this library [C:\Sources\bitcoin\build_msvc\libbitcoin_util\libbitcoin_util.vcxproj]


"C:\Sources\bitcoin\build_msvc\bitcoin.sln" (default target) (1) ->
"C:\Sources\bitcoin\build_msvc\bitcoind\bitcoind.vcxproj" (default target) (6) ->
"C:\Sources\bitcoin\build_msvc\libbitcoin_wallet\libbitcoin_wallet.vcxproj" (default target) (10) ->
(ClCompile target) ->
  C:\Sources\bitcoin\src\wallet\wallet.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj [C:\Sources\bitc
oin\build_msvc\libbitcoin_wallet\libbitcoin_wallet.vcxproj]

    3 Warning(s)
    1 Error(s)
```

Probably easy to fix, will check later.

### PR DESCRIPTION
**Pull request by NicolasDorier**
This fix a build error in Debug mode #16245

### CODE REVIEW

**Comment by NicolasDorier** [comment]
[bigobj doc](https://docs.microsoft.com/en-us/cpp/build/reference/bigobj-increase-number-of-sections-in-dot-obj-file?view=vs-2019) and the reason why it happens in [debug only](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1128?view=vs-2019).

**Comment by sipsorcery** [comment]
tACK eb832cd3b46a0f9abff238dfa780e4e9df05fdb7

As an aside I do get this issue with a VS2017 x64 debug built but NOT with a VS2019 x64 debug build so looks like something has been improved.

TLDR;

The root cause of the problem seems to be that when the compiler does not optimise the code the compilation fails with the "exceeded object file format error". The /Gy flag is not the cause instead it's the use of /Od (don't optimise to make debugging easier) and the lack of /O1 or /O2.

The command line below are the standard options before applying this fix.
```
"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\CL.exe" /c /I..\..\src /I..\..\src\univalue\include /I..\..\src\secp256k1\include /I..\..\src\leveldb\include /I..\..\src\leveldb\helpers\memenv /I"c:\Tools\vcpkg\installed\x64-windows-static\include" /ZI /JMC /nologo /W3 /WX /diagnostics:classic /sdl /Od /D _UNICODE /D UNICODE /D ZMQ_STATIC /D NOMINMAX /D WIN32 /D HAVE_CONFIG_H /D _CRT_SECURE_NO_WARNINGS /D _SCL_SECURE_NO_WARNINGS /D _CONSOLE /D _WIN32_WINNT=0x0601 /D _DEBUG /Gm- /EHsc /RTC1 /MTd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Debug\libbitcoin_wallet\wallet_wallet.obj" /Fd"x64\Debug\libbitcoin_wallet\libbitcoin_wallet.pdb" /Gd /TP /wd4018 /wd4221 /wd4244 /wd4267 /wd4715 /wd4805 /FC /errorReport:queue /utf-8 ..\..\src\wallet\wallet.cpp
c:\Dev\github\sipsorcery_bitcoin\src\wallet\wallet.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj
```

It can be fixed with `/bigobj` as per this PR or by replacing `/Od` with `/O1` or `/O2`. The `/bigobj` fix is the right option since it will keep the object files in an unoptimised state and more suitable for debugging.
